### PR TITLE
[feat] Session Context

### PR DESCRIPTION
**Pull request by itsliterallymonique**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: api/supabase/queries/auth.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { UUID } from "crypto";
```
Comment (me-liu) [inline api/supabase/queries/auth.ts]: I believe delete this file as it'll be merged in later

File: utils/SessionProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+export function SessionProvider({ children }: { children: React.ReactNode }) {
+  const [userId, setUserId] = useState<string | null>(null);
+  const [sessionLevelContext, setSessionLevelContext] =
+    useState<SessionLevelContext | null>(null);
+  const [participantSessionContext, setParticipantSessionContext] =
+    useState<ParticipantSessionContext | null>(null);
+
+  return <div></div>;
```
Comment (me-liu) [inline utils/SessionProvider.tsx]: I believe you should return
```
return (
    <SessionContext.Provider
      value={{ userId, sessionLevelContext, participantSessionContext }}>
      {children}
    </SessionContext.Provider>
  );```

File: utils/SessionProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    .from("session")
+    .select("is_async")
+    .eq("session_id", user_session_id)
+    .single();
+
+  if (error) {
+    console.error("Error fetching session by session_id:", error);
+  }
+
+  const is_async: boolean = data as unknown as boolean;
```
Comment (me-liu) [inline utils/SessionProvider.tsx]: `const is_async = data?.is_async ?? false;`

File: utils/SessionProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+// Participant-specific info from profile table
+export interface ParticipantSessionContext {
+  roleId: string;
+  phaseId: string;
+  isFinished: boolean | null;
+}
+
+export async function getUserId(session_id: string): Promise<UUID> {
+  const { data, error } = await supabase
+    .from("sessions")
```
Comment (me-liu) [inline utils/SessionProvider.tsx]: should be session

File: utils/SessionProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+  if (error) {
+    console.error("Error fetching session by session_id:", error);
+  }
+
+  const is_async: boolean = data as unknown as boolean;
+  return is_async;
+}
+
+export async function getSessionById(
```
Comment (me-liu) [inline utils/SessionProvider.tsx]: yeah I'm pretty sure you can't use useState in async functions. I think you just need to change this to a regular function (no async) and then add a useEffect inside with [sessionId] as the dependency

ie
```
export function useSessionById {
    SessionLevelContext inital states
    useEffect(()=>
        await everything and set the rest of the session level context
     } [sessionId]
```

File: utils/SessionProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+export async function getTemplateId(user_session_id: UUID): Promise<UUID> {
+  const { data, error } = await supabase
+    .from("session")
+    .select("template_id")
+    .eq("session_id", user_session_id)
+    .single();
+
+  if (error) {
```
Comment (me-liu) [inline utils/SessionProvider.tsx]: in these async functions, can you check if there isn't data

like if (error | !data)
